src/dbus_api.rs

```diff
@@ -185,35 +183,68 @@ fn remove_devs(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
     Ok(vec![m.msg.method_return().append3("/dbus/cache/path", 0, "Ok")])
 }
 
-fn create_dbus_pool<'a>(dbus_context: Rc<RefCell<DbusContext>>) -> dbus::Path<'a> {
+fn create_dbus_pool<'a>(dbus_context: &Rc<RefCell<DbusContext>>) -> dbus::Path<'a> {
 
     let f = Factory::new_fn();
 
-    let create_filesystems_method = f.method(CREATE_FILESYSTEMS, (), create_filesystems);
+    let create_filesystems_method = f.method(CREATE_FILESYSTEMS, (), create_filesystems)
+        .in_arg(("volumes", "a(sss)"))
+        .out_arg(("results", "a(oqs)"))
+        .out_arg(("return_code", "q"))
+        .out_arg(("return_string", "s"));
 
-    let destroy_filesystems_method = f.method(DESTROY_FILESYSTEMS, (), destroy_filesystems);
+    let destroy_filesystems_method = f.method(DESTROY_FILESYSTEMS, (), destroy_filesystems)
+        .in_arg(("volumes", "a(sqs)"))
+        .out_arg(("results", "a(oqs)"))
```

Comment:
Let's change results to "a(qs)" since the object path is now obsolete and we get the analogy of returning a just-freed pointer.